app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,28 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+
```

Comment:
- [ ] It would be better if a post is created on behalf of the currently logged in user (current_user) of the app.
eg: `@post = @current_user.posts.new(post_params)`

Reply:
I'm already creating for the current_user(first user according to project requirements). Otherwise, we wouldn't be able to see that post is that user page. Because in the example code, we are trying to create an object from an array. @current_user.posts refers to array

Reply:
No, it doesn't refer to an array but to the association between user and post models.

Creating a post as suggested in the example will automatically add the user_id attribute in the new post object. 

It makes easier to create any object that has user_id as a foreign key (eg: comment and like)

I hope that it makes sense to you.🙂

Reply:
I tried to use example code but it gave me error :(

Reply:
Okay, now the posts are created on that particular user.